src/PROPOSAL/detail/PROPOSAL/crosssection/parametrization/Photoproduction.cxx

```diff
@@ -16,6 +16,18 @@ size_t crosssection::Photoproduction::GetHash(const ParticleDef&, const Medium &
     return combined_hash;
 }
 
+double crosssection::Photoproduction::GetLowerEnergyLim(const ParticleDef&, const Medium& medium, cut_ptr) const {
+    double min_cut_off = INF;
+    for (const auto& c : medium.GetComponents())
+        min_cut_off = std::min(min_cut_off, GetCutOff(c));
+    return min_cut_off;
+};
+
+double crosssection::Photoproduction::GetCutOff(const Component& comp) const {
+    auto m_N = comp.GetAverageNucleonWeight();
+    return MPI + MPI * MPI / (2. * m_N);
```

Comment:
Maybe it is better to show, where this comes from, which is `crosssection::Photonuclear::GetKinematicLimits`

Reply:
at least
initially, I thought it can be defined in one place and then used in multiple applications, but maybe the overhead is too large overhead